Azure/sonic-mgmt
# [dualtor] Implement y_cable_simulator client

Signed-off-by: bingwang <bingwang@microsoft.com>


### Description of PR


Summary:
Fixes # (issue)
This PR is to introduce a simulator for y_cable for testing. 
The simulator driver is injected into ```pmon``` container in ```deploy_mg``` for ```dualtor``` topo and hook the physical y_cable driver. The simulator interact with simulator driver running on vmhost server, and control the behavior of y_cable with open-flow.

### Type of change




### Approach
#### What is the motivation for this PR?
This commit contains the code for y_cable_simulator client. 

#### How did you do it?
The simulator driver is injected to ```pmon``` container in 'deploy_mg' for dualtor topo and hook the physical y_cable driver. The simulator interact with simulator driver running on vmhost server, and control the behavior of y_cable with open-flow.

#### How did you verify/test it?
Verified on a dual-tor topo.
1. Verify the simulator is generated and injected to both DUTs in testbeds when deploy-mg.
2. Verify the simulator works as expected.

#### Any platform specific information?
No.

#### Supported testbed topology if it's a new test case?
Only applicable to dualtor topo.

### Documentation
